### ISSUE TITLE
Partitioning For Disc 8 Broken

### ISSUE THREAD

**Issue by progtronix** [FDOS/fdisk#73]
yea i know, why would anyone in 2023 bang 8 drives in dos, but ya know, vmware lets me and i needed to detect disks and partitions in dos so i used /dump to list all disks then looped backwards through the numbers using /info %i% to get partitions for found disks in batch script. thats when i noticed fdisk doesnt apply a partitions to disk 8 (even if i jumble the disks so either scsi or ide lands in disk 8, neither will partition). It will read that its there, it will delete them, but it just wont apply them and so my batch always reads a disk with empty partition right after i partitioned them all. i tested this in win98 boot disk (so 7.10) and FDOS 1.3 which i think still ships with 1.3.4 but this happens in all versions i tested from 1.3.4 to 1.3.10.

if i use microsofts fdisk's with 'fdisk 8/pri:65536' , it makes the full partition and then ffdisk can do anything after that.

tried in vmware with 4-ide(1gb) drives and 4-scsi(2gb) drives, then on w98 bootdisk i tried with both scsi and ide in disc 8 position, then i switched over a mini 1.3 fdos floppy and tried the same and it also failed to create the disc, on fdos i only tried 1 dirve (not sure if it was ide or scsi at that time and i did not try ms's fdisk on fdos as i assumed for ffdisk to have the best chance to work in fdos and after the first fail i assumed the results would be the same),  i even tried the interactive menu and that has the same results which it can do anything but apply a partition to disc 8

**Comment by boeckmann** [comment]
Hi @progtronix. Thanks for the detailed report. This indeed sounds like a bug in Free FDISK. I will try to deal with it in the next days. May I come back to you when a new build is ready so you can check if this is fixed?

**Comment by boeckmann** [comment]
@progtronix I think I already found the bug. While not having the time to test it by myself today (have to setup a VM etc.), I uploaded a test build in case you want to test it.

https://nextcloud.iww.rwth-aachen.de/index.php/s/8Ha79cyR9SN9qZY

**Comment by progtronix** [comment]
thanks for the quick response, i will get straight on this and let you know soon, thanks.

**Comment by progtronix** [comment]
yes that seems to have fixed the problem, thanl you for getting on this and for the fast fix, i can now drop 60kb and remove the need for MS fdisk and an additional check and goto lol.

Much Appreciated.

**Comment by boeckmann** [comment]
Unlike the release, the test build is not UPXed. The test build is 75k, the release around 40k. Either you UPX the test build or wait for a point release. I will package it the next days as 1.3.11.

Thanks again for reporting this! This bug may have existed for a long time. Like you said, EIGHT drives :)

**Comment by progtronix** [comment]
im not sure on what that is or how to do it, so its easier for me to just wait for the release version right now and continue with other things, im compressing it in a cab anyway and it seems to have turned out roughly the same size so its no problem being slightly larger until the next release, no rush, just glad its fixed, its only because i sit spend way to long testing all ranges just to know and be sure and to learn, or eventually... there will be an off by one problem lol.

again much appreciated

**Comment by boeckmann** [comment]
fixed by 96673f1

**Comment by mateuszviste** [comment]
> yes that seems to have fixed the problem, thank you for getting on this and for the fast fix, i can now drop 60kb and remove the need for MS fdisk and an additional check and goto lol.

If you are looking for savings, you could compile FDISK in "LITE" mode, it's 30% smaller with the same features, but without the interactive UI. I wonder if it would make sense that FDISK releases come in two versions, lite vs normal...

**Comment by boeckmann** [comment]
> If you are looking for savings, you could compile FDISK in "LITE" mode, it's 30% smaller with the same features, but without the interactive UI. I wonder if it would make sense that FDISK releases come in two versions, lite vs normal...

I can put two binaries in the Github release. The full FDISK and a command line only FDISKCMD. For installation boot disk, I think it is more user friendly to give them the full interface (not everyone wants to deal with command line).

**Comment by progtronix** [comment]
i think having separate bins would be great for this project, i feel people mostly need one or the other and rarely hit the need for both, setups are usually scripted menu options that execute cmd options rather than showing a page full of info + 10 keyboard inputs to memorise to get through the next interactive section in case the person has never seen the interactive menu before so its easier to show a page with dedicated options in constant view and execute cmd's and never enter interactive. or really your presenting a memory game while people are trying to partition a drive for the first time and if its me, i wont remember 2 commands and will have to reset and take a picture lol.

i think if people making setup disks want to be safe and have space left at the end for nothing they may include the full version anyways just to be safe and maybe offer it as a menu option if the person already knows how to use it. but in my case i would rather pack the last with drivers and things to help broaden the use so in the end i expect it always to be full so its all about how much i can trim with small programs until i get to the end and start adding extra drivers and programs.

long story short, 2 bins would be lovely :)

**Comment by mateuszviste** [comment]
>  so its easier to show a page with dedicated options in constant view and execute cmd's and never enter interactive.

That's what I think, too. And it's exactly what I plan to do with the SvarDOS installer - instead of the full FDISK UI version, the installer will only perform command-line calls to the cmdline-only FDISK (<https://github.com/SvarDOS/bugz/issues/26>). I felt your need seem similar, hence a proper LITE release of FDISK would be indeed great.
